backend/tests/vehiclePropertyControllerTest.js

```diff
@@ -0,0 +1,151 @@
+          response.body.should.be.a('object');
+          response.body.success.should.be.true;
+          vehicle_prop_2_id = response.body.payload._id;
+          vehicle_prop_2 = response.body.payload;
+          done();
+        });
+    });
+  });
+
+  //The below test checks if we can fetch a vehicle property successfully
+  describe('Get Vehicle Properties API Test', () => {
+    it('should return 200 when vehicle properties are returned', (done) => {
+      chai.request(app)
+        .get("/api/v1/dealerships/618b3bf134f07d9a91c32a1b/vehicles/properties")
+        .set('authorization', 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI2MThhYWNmNDVjZGM3NWI4Mjg4ZWI5YjUiLCJpYXQiOjE2MzgxMzc1ODUsImV4cCI6MTY0MDcyOTU4NX0.gv4Q-vPm-MqfHVsY7BYnBThxBI3bHZtUC5JtukHp340')
+        .end( (err, response) => {
+          response.should.have.status(200);
+          response.body.should.be.a('object');
+          response.body.success.should.be.true;
+          done();
+        });
+    });
+  });
+
+  //The below test checks if we can update a vehicle property successfully
+  describe('Update Vehicle Property API Test', () => {
+    it('should return 200 when a vehicle property is updated', (done) => {
+      chai.request(app)
+        .put("/api/v1/dealerships/618b3bf134f07d9a91c32a1b/vehicles/properties/" + vehicle_prop_1_id)
+        .set('authorization', 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI2MThhYWNmNDVjZGM3NWI4Mjg4ZWI5YjUiLCJpYXQiOjE2MzgxMzc1ODUsImV4cCI6MTY0MDcyOTU4NX0.gv4Q-vPm-MqfHVsY7BYnBThxBI3bHZtUC5JtukHp340')
+        .send({
+            'label': 'test_updated'
+        })
+        .end( (err, response) => {
+          response.should.have.status(200);
+          response.body.should.be.a('object');
+          response.body.success.should.be.true;
+          response.body.payload.label.should.be.equal("test_updated");
+          done();
+        });
+    });
+  });
+
+  //The below test checks if we can update the position of all vehicle properties successfully
+  describe('Update Vehicle Property Positions API Test', () => {
+    it('should return 200 when vehicle properties positions are updated successfully', (done) => {
+        let newProperties = [];
+        newProperties.push(vehicle_prop_2);
+        newProperties.push(vehicle_prop_1);
+        console.log(vehicle_prop_2._id);
+        console.log(vehicle_prop_1._id);
+        chai.request(app)
+        .put("/api/v1/dealerships/618b3bf134f07d9a91c32a1b/vehicles/properties")
+        .set('authorization', 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI2MThhYWNmNDVjZGM3NWI4Mjg4ZWI5YjUiLCJpYXQiOjE2MzgxMzc1ODUsImV4cCI6MTY0MDcyOTU4NX0.gv4Q-vPm-MqfHVsY7BYnBThxBI3bHZtUC5JtukHp340')
+        .send({
+            'properties': newProperties
+        })
+        .end( (err, response) => {
+          response.should.have.status(200);
+          response.body.should.be.a('object');
+          response.body.success.should.be.true;
```

Comment:
You can verify that the first vehicle property in the response array has position 1, that the second has position 2, etc.
`response.body.payload[0].position.should.be.equal(1)`
`response.body.payload[1].position.should.be.equal(2)`

Reply:
I have optimized the above test in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/34/commits/2c1fa39cd96d281067f13e5628c2e5edf104968d)